Source/DataSources/ParticleSystemVisualizer.js

```diff
@@ -0,0 +1,242 @@
+/*global define*/
+define([
+        '../Core/AssociativeArray',
+        '../Core/BoundingSphere',
+        '../Core/Cartesian3',
+        '../Core/Color',
+        '../Core/defined',
+        '../Core/destroyObject',
+        '../Core/DeveloperError',
+        '../Core/Matrix4',
+        '../Scene/CircleEmitter',
+        '../Scene/ParticleSystem',
+        './BoundingSphereState',
+        './Property'
+    ], function(
+        AssociativeArray,
+        BoundingSphere,
+        Cartesian3,
+        Color,
+        defined,
+        destroyObject,
+        DeveloperError,
+        Matrix4,
+        CircleEmitter,
+        ParticleSystem,
+        BoundingSphereState,
+        Property) {
+    'use strict';
+
+    var defaultStartScale = 1.0;
+    var defaultEndScale = 1.0;
+    var defaultStartColor = Color.WHITE;
+    var defaultEndColor = Color.WHITE;
+    var defaultRate = 5.0;
+    var defaultMinWidth = 16.0;
+    var defaultMinHeight = 16.0;
+    var defaultMaxWidth = 16.0;
+    var defaultMaxHeight = 16.0;
+    var defaultLifeTime = Number.MAX_VALUE;
+    var defaultLoop = true;
+    var defaultEmitterModelMatrix = Matrix4.IDENTITY;
+    var defaultEmitter = new CircleEmitter({radius: 0.5});
+    var defaultMinSpeed = 5.0;
+    var defaultMaxSpeed = 5.0;
+    var defaultMinLife = 5.0;
+    var defaultMaxLife = 5.0;
+
+    var modelMatrixScratch = new Matrix4();
+
+    /**
+     * A {@link Visualizer} which maps {@link Entity#model} to a {@link Model}.
+     * @alias ModelVisualizer
+     * @constructor
+     *
+     * @param {Scene} scene The scene the primitives will be rendered in.
+     * @param {EntityCollection} entityCollection The entityCollection to visualize.
+     */
+    function ParticleSystemVisualizer(scene, entityCollection) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(scene)) {
+            throw new DeveloperError('scene is required.');
+        }
+        if (!defined(entityCollection)) {
+            throw new DeveloperError('entityCollection is required.');
+        }
+        //>>includeEnd('debug');
+
+        entityCollection.collectionChanged.addEventListener(ParticleSystemVisualizer.prototype._onCollectionChanged, this);
+
+        this._scene = scene;
+        this._primitives = scene.primitives;
+        this._entityCollection = entityCollection;
+        this._particleHash = {};
+        this._entitiesToVisualize = new AssociativeArray();
+        this._onCollectionChanged(entityCollection, entityCollection.values, [], []);
+    }
+
+    /**
+     * Updates models created this visualizer to match their
```

Comment:
Copy and pasted the ModelVisualizer to make the ParticleSystemVisualizer and didn't update all the comments ;)  Fixed now.